db/model/aspect.js

```diff
@@ -217,6 +252,27 @@ module.exports = function aspect(seq, dataTypes) {
           );
         }
 
+        if (inst.changed('tags')) {
+          return new seq.Promise((resolve, reject) => {
+            inst.getSamples()
+            .each((samp) => {
+              return common.sampleAspectAndSubjectArePublished(seq, samp)
+              .then((published) => {
+                if (published) {
+                  return common.augmentSampleWithSubjectAspectInfo(seq, samp)
+                  .then((s) => {
+                    common.publishChange(s, sampleEventNames.add);
+                  });
+                }
+
+                return seq.Promise.resolve(true);
+              });
+            })
+            .then(() => resolve(inst))
+            .catch(reject);
```

Comment:
@iamigo do we have a test that asserts on unpublishing an aspect, it loses all of its samples?